qiskit/chemistry/core/hamiltonian.py

```diff
@@ -306,11 +305,12 @@ def _process_z2symmetry_reduction(self, qubit_op, aux_ops):
                     aux_ops[i] = None  # Discard since no meaningful measurement can be done
 
             if self._z2symmetry_reduction == 'auto':
-                hf_state = HartreeFock(num_orbitals=self._molecule_info[self.INFO_NUM_ORBITALS],
-                                       qubit_mapping=self._qubit_mapping,
-                                       two_qubit_reduction=self._two_qubit_reduction,
-                                       num_particles=self._molecule_info[self.INFO_NUM_PARTICLES])
-                z2_symmetries = Hamiltonian._pick_sector(z2_symmetries, hf_state.bitstr)
+                from ..circuit.library.initial_states.hartree_fock import _build_bitstr
```

Comment:
To me this does not look so good running off a private member. Maybe the better thing to do is to have some util somewhere that can build the bitstr so that the HF state or anything else like this can use it. If nothing it should be at least a public method of HFState if its preferable to keep it there for now.

Reply:
I thought about these two options too, but the problem with these are, that 
* to extract the bitstring from the circuit we'd have to simulate the circuit and retrieve the bitstring, which is exponentially expensive 
* having the property for the bitstring on it is a bit weird because initial states don't generally have to have a bitstring
* also we're creating a circuit but only care about the bitstring it is built from, this seems like an inverted workflow

How about changing this from a private function to a public method? E.g. `hartree_fock_bitstring`?

Reply:
A public method seems fine - the main suggestion was some sort of utility method/module somewhere outside of the circuit that the circuit could use. We have no util folder at present, hence other things, like MP2Info are in the root. Adding another file there for one method seems too much. Hence having it there with the circuit as a public method seems ok for now. `hartree_fock_bitstring` or `hf_bitstring` seems fine to me.